presto-docs/src/main/sphinx/optimizer/history-based-optimization.rst

```diff
@@ -0,0 +1,97 @@
+
+Examples of HBO
+---------------
+
+An example of a query plan with HBO statistics is shown below. For a plan node, the estimation statistics will show source `HistoryBasedSourceInfo` when the statistics are from HBO.
+
+.. code-block:: text
+
+         Fragment 1 [HASH]                                                                                                                                            >
+             Output layout: [orderpriority, count]                                                                                                                    >
+             Output partitioning: SINGLE []                                                                                                                           >
+             Stage Execution Strategy: UNGROUPED_EXECUTION                                                                                                            >
+             - Project[PlanNodeId 392][projectLocality = LOCAL] => [orderpriority:varchar(15), count:bigint]                                                          >
+                     Estimates: {source: HistoryBasedSourceInfo, rows: 5 (117B), cpu: ?, memory: ?, network: ?}                                                       >
+                 - Aggregate(FINAL)[orderpriority][$hashvalue][PlanNodeId 4] => [orderpriority:varchar(15), $hashvalue:bigint, count:bigint]                          >
+                         Estimates: {source: HistoryBasedSourceInfo, rows: 5 (117B), cpu: ?, memory: ?, network: ?}                                                   >
+                         count := "presto.default.count"((count_8)) (1:50)                                                                                            >
+                     - LocalExchange[PlanNodeId 354][HASH][$hashvalue] (orderpriority) => [orderpriority:varchar(15), count_8:bigint, $hashvalue:bigint]              >
+                         - RemoteSource[2] => [orderpriority:varchar(15), count_8:bigint, $hashvalue_9:bigint]
+
+    
+Optimizations using HBO
+-----------------------
+
+DetermineJoinDistributionType & DetermineSemiJoinDistributionType
+-----------------------------------------------------------------
+These two optimizations decide whether to do broadcast or repartition for a join. They use the size of the probe and build input in optimization.
+* Data size recorded from history queries will be used when HBO is enabled.
+* Statistics from cost models are used when HBO statistics are not available or HBO is disabled.
+
+ReorderJoins
+------------
+This optimization reorders the join order based on the size of input and output. Data size recorded from history queries will be used when HBO is enabled.
+
+PushPartialAggregationThroughExchange
+-------------------------------------
+This optimization decides whether to split an aggregation into partial and final aggregations. HBO has special session properties to control it.
```

Comment:
```suggestion
This optimization decides whether to split an aggregation into partial and final aggregations. 

```
Need a line after this text for the two items in the unordered list to build correctly.